### PR TITLE
Add git branch and commit info to footer

### PR DESCRIPTION
**Pull request by Malet**
This needs to be set by GIT_COMMIT and GIT_BRANCH envvars in production

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -22,6 +31,24 @@ config :logger, :console,
...
   metadata: [:request_id]
 
+git_branch =
+  Config.get_env("GIT_COMMIT") ||
+    System.cmd("git", ~w(rev-parse --abbrev-ref HEAD))
+    |> elem(0)
+    |> String.trim
+
+git_commit =
+  Config.get_env("GIT_COMMIT") ||
```
Comment (IanLuites) [inline config/config.exs:41]: In production mode this will always be `"${GIT_COMMIT}"`.

File: config/config.exs [hunk 2]
```diff
@@ -22,6 +31,24 @@ config :logger, :console,
...
+    System.cmd("git", ~w(rev-parse HEAD))
```
Comment (IanLuites) [inline config/config.exs:42]: This crashes compilation on systems without `git`.

I did it like this once before: [Status.System.commit/0](https://github.com/shinyscorpion/elixir-status/blob/prototype/pre0/lib/status/system.ex#L5) and is now used in multiple projects.

We can do something similar or we can include the `:status` package and I'll add `branch/0` to that.

**Review by IanLuites — COMMENTED** [review]
👍

**Comment by IanLuites** [comment]
Does this need to be set by the environment variables in production though? If we're going to build a release with distillery, won't it make sense to compile in the commit and branch at compile time. (if available ofc, this is what the `:status` package does.)

File: config/config.exs (hunk 2 above)
Comment (Malet) [reply config/config.exs:42]: I'm ok with the crashes, our dev machines will have it, and at build time it should fail if we haven't set the environment variable

**Pull request merged**